### DISCUSSION TITLE
edk2 setup build failed

### DISCUSSION THREAD

**Discussion by tushar95mahawar** [tianocore/edk2#3629] [General]
Hello everyone. I am beginner  i wan to build edk setup but build process is failed.

C:\edk2>build
Build environment: Windows-post2008Server-6.2.9200
Build start time: 11:07:27, Nov.11 2022

WORKSPACE        = c:\edk2
EDK_TOOLS_PATH   = c:\edk2\basetools
EDK_TOOLS_BIN    = c:\edk2\basetools\bin\win32
CONF_PATH        = c:\edk2\conf
PYTHON_COMMAND   = C:\Python27\Tools\Scripts\python.exe

Architecture(s)  = X64

Processing meta-data Build target     = DEBUG
.Toolchain        = VS2015x86

Active Platform          = c:\edk2\MdeModulePkg\MdeModulePkg.dsc
....

- Failed -
Build end time: 11:07:32, Nov.11 2022
Build total time: 00:00:05

**Comment by lersi** [comment]
python 2 is no longer supported by the build system, please use python 3.7 or later.

I mad a project for starters that explains the build system and provides easy compilation.
project link: [https://github.com/lersi/edk2_project_template](https://github.com/lersi/edk2_project_template)

**Reply by tushar95mahawar** [reply]
!!! ERROR !!! Binary C tools are missing. They are required to be built from BaseTools Source.

I'm facing this error also when i gave command edksetup.bat

**Reply by lersi** [reply]
try using the command `git submodule update --init` in edk2's root directory

**Reply by tushar95mahawar** [reply]
still not progress

**Comment by spbrogan** [comment]
The guide has been recently updated.  I would suggest you follow this. https://github.com/tianocore/tianocore.github.io/wiki/How-to-Build-With-Stuart
